subprojects/logging/src/main/java/org/gradle/internal/logging/sink/OutputEventTransformer.java

```diff
@@ -101,9 +101,13 @@ public void onOutput(OutputEvent event) {
                     outputEvent = outputEvent.withBuildOperationId(mappedId);
                 }
             }
-            listener.onOutput(outputEvent);
+            invokeListener(outputEvent);
         } else {
-            listener.onOutput(event);
+            invokeListener(event);
         }
     }
+    private synchronized void invokeListener(OutputEvent event) {
```

Comment:
Looking at the changes more in detail, this does not seem correct.

The synchronization is not on the same `lock` object as in `OutputEventRenderer`.

You should pass the `lock` instance from that type in this type and synchronize on it to restore the behavior.